genesyscloud/task_management_worktype/genesyscloud_task_management_worktype_proxy.go

```diff
@@ -158,8 +165,8 @@ func getAllTaskManagementWorktypeFn(ctx context.Context, p *taskManagementWorkty
 	return &allWorktypes, nil
 }
 
-// getTaskManagementWorktypeIdByNameFn is an implementation of the function to get a Genesys Cloud task management worktype by name
-func getTaskManagementWorktypeIdByNameFn(ctx context.Context, p *taskManagementWorktypeProxy, name string) (id string, retryable bool, err error) {
+// Looksup a worktype by name
+func getWorkType(name string, p *taskManagementWorktypeProxy) (*platformclientv2.Worktype, bool, error) {
```

Comment:
The two data sources do almost the same actions.  The difference is that the data original data source we retrieved only the work type id.  The new data source retrieves the work type and then cycles through to match on the status.  Since they are doing the same API calls under the covers, I  refactored the calls out to our API to be in a single shared method called getWorkType.